x/utils/fn/fn.go

```diff
@@ -0,0 +1,307 @@
+package fn
```

Comment:
what tyler said. also I'm generally opposed to adding in unused code, if for no other reason other than audits get more expensive for no reaason